internal/controller/nonadminbackup_controller.go

```diff
@@ -97,11 +144,279 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+// to reflect that deletion has been initiated, and triggers the actual deletion if needed.
+//
+// Parameters:
+//   - ctx: Context for managing request lifetime.
+//   - logger: Logger instance for logging messages.
+//   - nab: Pointer to the NonAdminBackup object being processed.
+//
+// Returns:
+//   - bool: true if reconciliation should be requeued, false otherwise
+//   - error: any error encountered during the process
+func (r *NonAdminBackupReconciler) setStatusAndConditionForDeletionAndCallDelete(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	updatedPhase := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseDeleting)
+	updatedCondition := meta.SetStatusCondition(&nab.Status.Conditions,
+		metav1.Condition{
+			Type:    string(nacv1alpha1.NonAdminConditionDeleting),
+			Status:  metav1.ConditionTrue,
+			Reason:  "DeletionPending",
+			Message: "backup accepted for deletion",
+		},
+	)
+	if updatedPhase || updatedCondition {
+		if err := r.Status().Update(ctx, nab); err != nil {
+			logger.Error(err, statusUpdateError)
+			return false, err
+		}
+		logger.V(1).Info("NonAdminBackup status marked for deletion")
+		return true, nil
+	}
+	if nab.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.V(1).Info("Marking NonAdminBackup for deletion", nameString, nab.Name)
+		if err := r.Delete(ctx, nab); err != nil {
+			logger.Error(err, "Failed to call Delete on the NonAdminBackup object")
+			return false, err
+		}
+		return true, nil // Requeue to allow deletion to proceed
+	}
+	return false, nil
+}
+
+// setStatusForDirectKubernetesApiDeletion updates the status and conditions when a NonAdminBackup
+// is deleted directly through the Kubernetes API. Only updates status and conditions
+// if the NAB finalizer exists.
+//
+// Parameters:
+//   - ctx: Context for managing request lifetime
+//   - logger: Logger instance
+//   - nab: NonAdminBackup being deleted
+//
+// Returns:
+//   - bool: true if status was updated
+//   - error: any error encountered
+func (r *NonAdminBackupReconciler) setStatusForDirectKubernetesApiDeletion(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	if controllerutil.ContainsFinalizer(nab, constant.NabFinalizerName) {
+		updatedPhase := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseDeleting)
+		updatedCondition := meta.SetStatusCondition(&nab.Status.Conditions,
+			metav1.Condition{
+				Type:    string(nacv1alpha1.NonAdminConditionDeleting),
+				Status:  metav1.ConditionTrue,
+				Reason:  "DeletionPending",
+				Message: "backup deletion requires VeleroBackup deletion or finalizer removal",
+			},
+		)
+		if updatedPhase || updatedCondition {
+			if err := r.Status().Update(ctx, nab); err != nil {
+				logger.Error(err, statusUpdateError)
+				return false, err
+			}
+			logger.V(1).Info("NonAdminBackup status marked for deletion")
+			return true, nil
```

Comment:
here we can exit, no? If it enters reconcile again, it does not have anything else to do

Reply:
It does not, but I want to keep it same across all status updates, so whenever it succeeded or not we do want to either exit with error or reconcile so next time it enters reconciles there is no update and ends the reconcile.

This way we have same pattern for each status update, otherwise we may be introducing variations which may cause later some wired behavior.

@shubham-pampattiwar any opinion on that?

Reply:
Ok I removed that and added comment in the code, why.

Reply:
Done please check newest commit.